src/main/java/Duke.java

```diff
@@ -1,10 +1,124 @@
+import java.util.Scanner;
+
 public class Duke {
+
+    private static final String border = "==================================================";
+    private static final int MAX_LIST_CAPACITY = 100;
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Task[] tasks = new Task[MAX_LIST_CAPACITY];
+        printWelcomeMessage();
+
+        Scanner in = new Scanner(System.in);
+        String line = in.nextLine();
+
+        while (!line.equals("bye")) {
+
+            if (line.equals("list")) {
+                printList(tasks);
+
+            } else if (line.startsWith("done ")) {
+                int taskNumber = markTaskAsDone(tasks, line);
+                printDoneTaskMessage(tasks[taskNumber]);
+
+            } else if (line.startsWith("todo")) {
+                addToDo(tasks, line);
+                printAddTaskMessage(tasks[Task.taskCount - 1]);
+
+            } else if (line.startsWith("deadline")) {
+                addDeadline(tasks, line);
+                printAddTaskMessage(tasks[Task.taskCount - 1]);
+
+
+            } else if (line.startsWith("event")) {
+                addEvent(tasks, line);
+                printAddTaskMessage(tasks[Task.taskCount - 1]);
+
+            } else {
+                printErrorMessage();
+            }
+
+            line = in.nextLine();
+        }
+        printByeMessage();
+    }
+
+    private static void addDeadline(Task[] tasks, String line) {
+
+        String deadline = line.substring(line.lastIndexOf("deadline "));
+        int firstIndexOfByCommand = deadline.indexOf("/by ");
+        int lastIndexOfByCommand = deadline.lastIndexOf("/by ");
+
+        String description = deadline.substring(0, firstIndexOfByCommand).trim();
+        String by = deadline.substring(lastIndexOfByCommand + 4).trim();
```

Comment:
Perhaps 4 could be related to the "/at" string to avoid being a magic number? Could even make "/at" a constant so that you can change it easily.
```
private static final String PARAM_AT = "/at";
```